contracts/StreamManager.vy

```diff
@@ -252,3 +320,43 @@ def withdraw(creator: address, stream_id: uint256) -> uint256:
     log Withdrawn(creator, stream_id, funded_amount == withdrawal_amount, withdrawal_amount)
 
     return withdrawal_amount
+
+
+@external
+def batch_withdraw(
+    batches: DynArray[BatchWithdraw, MAX_BATCH_SIZE]
+    ) -> uint256:
+    """
+    @dev Withdraw from all streams for a given creator.
+    @param batches the batch of creators and associated streams
+    """    
+    total_amount: uint256 = 0
+    token: ERC20 = self.streams[batches[0].creator][batches[0].stream_ids[0]].token
+    assert token.address != empty(address), "token must be set"
+
+    for i in range(MAX_BATCH_SIZE):
+        if convert(i, uint256) >= len(batches):
+            break
+        creator: address = batches[i].creator
+        stream_ids: DynArray[uint256, MAX_BATCH_SIZE] = batches[i].stream_ids
+        for j in range(MAX_BATCH_SIZE):
+            if convert(j, uint256) >= len(stream_ids):
+                break
+            stream_id: uint256 = stream_ids[j]
```

Comment:
```suggestion
    for batch in batches:
        for stream_id in batch.stream_ids:
```

And then below just use `batch.creator` instead of `batch`